spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/core/CredentialsProviderAutoConfiguration.java

```diff
@@ -101,4 +112,18 @@ private static ProfileCredentialsProvider createProfileCredentialProvider(Profil
 		return ProfileCredentialsProvider.builder().profileName(profile.getName()).profileFile(profileFile).build();
 	}
 
+	private static boolean shouldCreateStsIdentityTokenCredentialsProvider() {
+		return ClassUtils.isPresent(STS_WEB_IDENTITY_TOKEN_FILE_CREDENTIALS_PROVIDER, null)
+			&& StringUtils.hasText(System.getenv(AWS_ROLE_ARN_ENV_VARIABLE))
+			&& StringUtils.hasText(System.getenv(AWS_WEB_IDENTITY_TOKEN_FILE_ENV_VARIABLE));
+	}
+
+	private static StsWebIdentityTokenFileCredentialsProvider createStsCredentialsProvider() {
+		return StsWebIdentityTokenFileCredentialsProvider.builder()
+			.stsClient(StsClient.builder().build())
+			.roleArn(System.getenv(AWS_ROLE_ARN_ENV_VARIABLE))
+			.webIdentityTokenFile(Path.of(System.getenv(AWS_WEB_IDENTITY_TOKEN_FILE_ENV_VARIABLE)))
+			.build();
```

Comment:
We should add:
`.asyncCredentialUpdateEnabled(stsProperties.isAsyncCredentialsUpdate())`

So users have options to run sync or async calls.